Upstream-Status: Backport

From ae13d96fa2a0612b6000f4b8f6ed9d3564035703 Mon Sep 17 00:00:00 2001
From: Michael Biebl <biebl@debian.org>
Date: Sun, 10 Apr 2011 12:54:53 +0000
Subject: Build hald-probe-video4linux on current kernels again

The hald-probe-video4linux prober supports both v4l1 and v4l2. Support for v4l1
has been removed from Linux kernel 2.6.38. Instead of disabling the prober
altogether, #ifdef the v4l1 parts when building on a newer kernel.

Signed-off-by: Martin Pitt <martin.pitt@ubuntu.com>
---
(limited to 'hald/linux/probing/probe-video4linux.c')

diff --git a/hald/linux/probing/probe-video4linux.c b/hald/linux/probing/probe-video4linux.c
index 7bc13e8..b055720 100644
--- a/hald/linux/probing/probe-video4linux.c
+++ b/hald/linux/probing/probe-video4linux.c
@@ -30,7 +30,9 @@
 #include <sys/types.h>
 #include <sys/time.h>
 #include <sys/ioctl.h>
+#ifdef HAVE_LINUX_VIDEODEV_H
 #include <linux/videodev.h>
+#endif
 #include <linux/videodev2.h>
 #include <errno.h>
 #include <fcntl.h>
@@ -50,7 +52,9 @@ main (int argc, char *argv[])
 	int ret = -1;
 	char *udi;
 	char *device_file;
+#ifdef HAVE_LINUX_VIDEODEV_H
 	struct video_capability v1cap;
+#endif
 	struct v4l2_capability v2cap;
 	LibHalContext *ctx = NULL;
 	LibHalChangeSet *cset;
@@ -107,7 +111,9 @@ main (int argc, char *argv[])
 			LIBHAL_FREE_DBUS_ERROR (&error);
 			libhal_device_add_capability (ctx, udi, "video4linux.radio", &error);
 		}
-	} else {
+	}
+#ifdef HAVE_LINUX_VIDEODEV_H
+	else {
 		HAL_DEBUG (("ioctl VIDIOC_QUERYCAP failed"));
 
 		if (ioctl (fd, VIDIOCGCAP, &v1cap) == 0) {
@@ -134,6 +140,7 @@ main (int argc, char *argv[])
 			HAL_DEBUG (("ioctl VIDIOCGCAP failed"));
 		}
 	}
+#endif
 
 	LIBHAL_FREE_DBUS_ERROR (&error);
 	libhal_device_commit_changeset (ctx, cset, &error);
--
cgit v0.9.0.2-2-gbebe
